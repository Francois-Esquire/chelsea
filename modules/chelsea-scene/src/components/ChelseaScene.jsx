import React from 'react';
import { Route } from '@americanexpress/one-app-router';

import Scene from './Scene';

const ChelseaScene = () => (
  <Scene />
);

// Read about childRoutes:
// https://github.com/americanexpress/one-app/blob/master/docs/api/modules/Routing.md#childroutes
ChelseaScene.childRoutes = () => ([
  <Route path="/" />,
]);

// Read about appConfig:
// https://github.com/americanexpress/one-app/blob/master/docs/api/modules/App-Configuration.md
if (!global.BROWSER) {
  // eslint-disable-next-line global-require
  ChelseaScene.appConfig = require('../appConfig').default;
}

export default ChelseaScene;
