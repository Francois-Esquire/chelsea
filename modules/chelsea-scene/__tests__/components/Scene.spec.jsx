import React from 'react';
import { render } from '@testing-library/react';

import ChelseaScene from '../../src/components/Scene';

describe('ChelseaScene', () => {
  it('default export should return a function', () => {
    expect(ChelseaScene).toBeInstanceOf(Function);
  });

  it('module should render correct JSX', () => {
    const { asFragment } = render(<ChelseaScene />);
    expect(asFragment()).toMatchSnapshot();
  });
});
