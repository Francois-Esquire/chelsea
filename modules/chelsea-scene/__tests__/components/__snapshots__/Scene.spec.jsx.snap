// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChelseaScene module should render correct JSX 1`] = `
<DocumentFragment>
  <div
    style="position: relative; width: 100%; height: 100%; overflow: hidden;"
  >
    <canvas
      style="display: block;"
    />
  </div>
</DocumentFragment>
`;
